lib/pure/strutils.nim

```diff
@@ -1374,6 +1354,47 @@ proc format*(formatstr: string, a: varargs[string, `$`]): string {.noSideEffect,
   result = newStringOfCap(formatstr.len + a.len)
   addf(result, formatstr, a)
 
+# Moved down here because the macros module requires cmpIgnoreStyle and format
+import macros
+
+proc parseEnum*[T: enum](s: string): T =
+  ## Parses an enum ``T``.
+  ##
+  ## Raises ``ValueError`` for an invalid value in `s`. The comparison is
+  ## done in a style insensitive way.
+  macro m: stmt =
+    result = newNimNode(nnkCaseStmt).add(newDotExpr(newIdentNode("s"),
```

Comment:
This is very difficult to read, how about a `dumpTree`-style explanatory comment or an example of the output code?